src/base/static/utils/geospatial-utils.js

```diff
@@ -0,0 +1,183 @@
+            [property.name]: property.fallbackValue,
+          }),
+          {},
+        );
+  },
+  aggregatePointsInBuffer: ({ placeGeometry, sourceFeatures, config }) => {
+    const bufferFeature = getBufferFeature(placeGeometry, config.buffer);
+
+    if (!bufferFeature) {
+      return {};
+    }
+
+    const pointsWithin = pointsWithinPolygon(
+      featureCollection(sourceFeatures),
+      bufferFeature,
+    );
+
+    if (config.aggregator.type === "totalCount") {
+      return {
+        [config.name]: pointsWithin.features.length,
+      };
+    } else if (config.aggregator.type === "countByUniquePropertyValues") {
+      // Return an object that counts the number of occurrences of each unique
+      // value of `config.aggregator.property` in the set of points in the
+      // buffer, producing an object like:
+      // {
+      //   propertyA: 3,
+      //   propertyB: 7,
+      //   ...
+      // }
+      const uniqueValues = pointsWithin.features.reduce(
+        (memo, feature) => ({
+          ...memo,
+          [feature.properties[config.aggregator.property]]: memo[
+            feature.properties[config.aggregator.property]
+          ]
+            ? ++memo[feature.properties[config.aggregator.property]]
+            : 1,
+        }),
+        {},
+      );
+
+      return {
+        [config.name]: uniqueValues,
+      };
+    } else {
+      return {
+        [config.name]: pointsWithin.features.length,
+      };
+    }
+  },
+  aggregatePolygonsOverlappingBuffer: ({
+    placeGeometry,
+    sourceFeatures,
+    config,
+  }) => {
+    const bufferFeature = getBufferFeature(placeGeometry, config.buffer);
+
+    if (!bufferFeature) {
+      return {};
+    }
+
+    sourceFeatures = sourceFeatures.reduce((memo, feature) => {
+      if (feature.geometry.type === "MultiPolygon") {
+        // Flatten any MultiPolygons to arrays of individual Polygon features,
+        // because `booleanOverlap` can only compare like feature types.
+        try {
+          feature = flatten(feature).features;
+        } catch (e) {
+          // eslint-disable-next-line no-console
+          console.error(e);
+
+          feature = [];
+        }
+      }
+
+      return memo.concat(feature);
+    }, []);
+
+    let overlappingFeatures = sourceFeatures.filter(feature => {
+      try {
+        return (
+          // Technically, we're doing more than an overlap comparison here.
+          // `boooleanOverlap` returns features which are strictly overlapping; we
+          // also want to return features which are entirely contained by the
+          // buffer.
+          booleanOverlap(bufferFeature, feature) ||
+          booleanContains(bufferFeature, feature)
+        );
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.error(e);
+
+        return false;
+      }
+    });
+
+    const propertyHashes = new Set();
+    overlappingFeatures = overlappingFeatures.filter(overlappingFeature => {
```

Comment:
nit: I think this would be more clearly written by using a `reduce` instead of a `filter` here, which can look like this:

```js
.reduce( ({ propertyHashes, features }, feature) => {
  const propertyHash = hash(feature.properties);
  if (propertyHashes.has(propertyHash)) {
    return {
      propertyHashes,
      features: [
        ...features,
        feature,
      ],
    }
  } else {
    propertyHashes.add(propertyHash);
    return {
      propertyHashes,
      features,
    }
  }
}, { propertyHashes: new Set(), features: [] }).features
```

I would prefer it since it reduces the amount of state we need to keep track of - like mutating `overlappingFeatures` and closing over `propertyHashes`.

But I guess it depends on how much functional purism you are willing to tolerate ;)